core/src/main/java/io/parsingdata/metal/Shorthand.java

```diff
@@ -99,154 +100,215 @@
 
     private Shorthand() {}
 
+    /** Instantiates a {@link Def} nested in a {@link Post}. */
     public static Token def(final String name, final SingleValueExpression size, final Expression predicate, final Encoding encoding) { return post(def(name, size, encoding), predicate); }
+
+    /** Instantiates a {@link Def} with {@code encoding = null} nested in a {@link Post}. */
```

Comment:
I would not mention the post part here, but simply reference that shorthand instead. I also thought that you did that... Am I reviewing too soon?

Reply:
Actually, I thought that referencing those helpers methods would create a kind of click-through jungle that would not really help users of Metal to understand what they need to know. In this case the JavaDoc says that what you end up with is a `Def` and a `Post`, with links to the explanations of what both do. And in cases where relevant arguments are filled in with a default, that is also mentioned here.

Reply:
I wonder if the Javadoc tags introduced in Java 8 would help here: `@implSpec` and `@implNote`.

```java
/**
 * Instantiates a {@link Def} that succeeds when the predicate succeeds.
 *
 * @imlpSpec Instantiates a {@link Def} with {@code encoding = null} nested in a {@link Post}.
 */
```

Reply:
Sounds like a good approach for a future extension!